pkgs/development/python-modules/ogmios/default.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  hatchling,
+  setuptools,
+  # Python deps
+  cachetools,
+  cardano-tools,
+  coloredlogs,
+  orjson,
+  pydantic,
+  websockets,
+}:
+
+buildPythonPackage rec {
+  pname = "ogmios";
+  version = "1.2.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    pname = "ogmios";
+    inherit version;
+    hash = "sha256-B9cIbKjdtMo46cYe3aU/9KzA0VbRXqBW6sEc0FOQ2ew=";
+  };
+
+  build-system = [
+    hatchling
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    cachetools
+    cardano-tools
+    coloredlogs
+    orjson
+    pydantic
+    websockets
+  ];
+
```

Comment:
```suggestion

  nativeCheckInputs = [
    pytestCheckHook
  ];

```

Reply:
And this one is tricky
- ogmios' tests depend on pycardano https://gitlab.com/viperscience/ogmios-python/-/blob/6cb1b913d72f8c3da4eb72824354c50420f0ec14/pyproject.toml#L39
- pycardano depends on ogmios https://github.com/Python-Cardano/pycardano/blob/main/pyproject.toml#L38

I don't know how that's supposed to work so I'll skip that